### DISCUSSION TITLE
3D EBSD reconstruction problem

### DISCUSSION THREAD

**Discussion by wujun-yin** [BlueQuartzSoftware/DREAM3DNX-Issues#2] [Q&A] [answered]
Dear all
  I am trying to use the DREAM3D to reconstruct 3D EBSD structures. However, I got a problem when I used a filter "Align Sections (Misorientation)", which is " FilterParameter 'Crystal Structures' Validation Error: Object does not exist at path 'DataContainer/CellEnsembleData/CrystalStructures' ". But I am sure that the CrystalStructure data was created and set correctly in the required input box. Did anyone have this problem before? Or have any comments to solve it?
  Thank you very much for you kind help!

Yin

**Comment by nyoungbq** [comment]
Hello Yin!

I think we will need more information to properly diagnose this problem. I have a few questions to help better gauge the issue at hand:
- What Operating System and Version are you using [Examples: Windows 10, Ubuntu Linux 22.04, MacOS BigSur]?
- Is there a way you can share your pipeline, or if its proprietary information replicate it in a simple example pipeline?
- Is it possible to provide input files to us (i.e. non proprietary data being used)?
- Just to be clear this is for DREAM3DNX (not DREAM3D 6.5 or 6.6)?
- Provided this is DREAM3DNX, what release candidate version are you running? This can be found by doing the following: 
1. Click on the Help menu in the top bar
2. Click on "About DREAM.3D..."
3. In the Popup window it should display something along the lines of "DREAM3DNX Version 7.0.0-RC-7"
4. The RC-Number is your release candidate

Without more information I can only offer the basic tips such as hover over the FilterParameter in question and make sure that the restrictions on component shape and type match the array you are trying to select. Array information can be found by hovering over the array name in the DataStructure Tab (which can be found in the top right of the GUI if you have a default layout). If your array name does not appear in the DataStructure Tab then chances are our Database does not have it stored. This could be an issue do to a malfunction of input and output filters (usually named import/read and export/write), in which case more information about your pipeline and input files will be needed to properly sort your problem.

I am sorry about the late response and I do hope this helps. I look forward to hearing back from you and sorting this issue out in a quick an concise manner!

Nathan

Edit: its important to note that the DataStructure tab will change from filter to filter depending on the arrays being created or removed in previous filters in the pipeline, so be sure to inspect the DataStructure while clicked on the filter you are trying to add the array in. In this case that would be the **Align Sections (Misorientation)** filter in your pipeline. With this filter selected you should be able to see _CrystalStructures_ in the DataStructure tab, verify the spelling is correct in the parameter because it is case sensitive and if spaces are used in the name they must be included. Typically the best way is just to drag the array name from the DataStructure view into the subsequent filter parameter.

**Comment by imikejackson** [comment]
Dear Yin, 
   What kind of files is the EBSD data coming from? Is it .ang or .ctf files? or are these just ASCI text/CSV files?

**Comment by Sam-RW-Taylor** [comment]
Dear all,
I have also been experiencing this issue. If I process my data as a single .ctf file then it all works as expected, but when I convert my stack to a .h5ebsd file then the CrystalStructures issues arises. This issue is also there for other filters that require CrystalStructures, so far I have only tested generate IPF colors and Neighbor orientation correlation. I have also tried generating the .h5ebsd data using version 6.5.171 and I can load the file on that version fine, but not on version 7.0.0 RC-6.
To cover questions not already covered:
I am using Windows 11
My pipeline is simply load the .h5ebsd file and then generate IPF colors, with default settings

Sam

**Reply by nyoungbq** [reply]
@Sam-RW-Taylor Hey Sam, 
I apologize for the very late response it seems I am having issues with my alert system on this repository. I have reached out to the rest of the team now to see if we can replicate and identify the cause. I will do my best to update you as I get more information, I am moving this to under investigation now! Thank you so much for reaching out and providing further information on the problem.

Nathan

**Reply by imikejackson** [reply]
@Sam-RW-Taylor Are you still experiencing this issue with the recent RC-8 that was posted?

**Comment by Sam-RW-Taylor** [comment]
@nyoungbq  and @imikejackson 
Hi both thank you for looking in to this. I still have this issue when I updated to RC-7 but I haven't updated to RC-8 so I don't know if the issue persists. 
I have however found a work around by writing a .dream3d file in the 6.5.171 version and then using that file for RC-7. I have also been able to use .dream3d files I have written within RC-7. Because of this I expect this issue is with how the .h5ebsd file is read within RC-6 and 7.

I hope this helps find the issue.

Sam

**Comment by imikejackson** [comment]
@Sam-RW-Taylor  This is what RC-8 looks like for the AlignSections. Does yours look like this? Specifically the DataStructure tab on the right side of the window?

![Screenshot 2023-12-26 at 2 17 24 PM](https://github.com/BlueQuartzSoftware/DREAM3DNX-Issues/assets/5182396/5f6173f5-cc0a-4a04-893b-f3cc57505c46)

**Comment by Sam-RW-Taylor** [comment]
@imikejackson The data structure looks pretty similar and the preflight happens without issue, then when the pipeline is run the error in the bottom of the first picture is shown.
![page01](https://github.com/BlueQuartzSoftware/DREAM3DNX-Issues/assets/151179001/9719ecd9-a845-4476-b991-554bb877a590)
When looking at the data array info for CrystalStrucutres loaded from a H5EBSD file and a Dream3d file I noticed that the tuple shape was different, could that be a cause of the issue?
![page01](https://github.com/BlueQuartzSoftware/DREAM3DNX-Issues/assets/151179001/1bdc6ed9-669f-4498-9429-1d1f616e2aca)

**Reply by imikejackson** [reply]
@Sam-RW-Taylor Sorry for the delay in responding. I would agree that something does not look correct unless the imported EBSD data really does have 6 phases. Would it be possible to get access to the data you are trying to analyze?

**Reply by Sam-RW-Taylor** [reply]
@imikejackson  Yes I would be able to share some data, I don't think I can share it here so could you please email me at st777@cam.ac.uk and then I can send you it directly.

**Reply by imikejackson** [reply]
@Sam-RW-Taylor Thank you for the data. I was able to reproduce the issue.  I will investigate more to figure out the issue. Thank you again for the data set.

**Comment by imikejackson** [comment, marked as answer]
@Sam-RW-Taylor https://github.com/BlueQuartzSoftware/simplnx/issues/862
